backend/pkg/api/router.go

```diff
@@ -239,16 +239,22 @@ func addValidatorDashboardRoutes(hs *handlers.HandlerService, publicRouter, inte
 
 	publicDashboardRouter := publicRouter.PathPrefix(vdbPath).Subrouter()
 	internalDashboardRouter := internalRouter.PathPrefix(vdbPath).Subrouter()
+
+	// endpoints which should be accessible even if dashboard is archived
+	archivalEndpoints := []endpoint{
+		{http.MethodDelete, "/{dashboard_id}", hs.PublicDeleteValidatorDashboard, hs.InternalDeleteValidatorDashboard},
+		{http.MethodPut, "/{dashboard_id}/archiving", hs.PublicPutValidatorDashboardArchiving, hs.InternalPutValidatorDashboardArchiving},
+	}
+	archivalRoutes := addEndpointsToRouters(archivalEndpoints, publicDashboardRouter, internalDashboardRouter)
+
 	// add middleware to check if user has access to dashboard
 	if !cfg.Frontend.Debug {
-		publicDashboardRouter.Use(hs.VDBAuthMiddleware, hs.ManageViaApiCheckMiddleware)
-		internalDashboardRouter.Use(hs.VDBAuthMiddleware)
+		publicDashboardRouter.Use(hs.VDBAuthMiddleware, hs.ManageViaApiCheckMiddleware, hs.GetMiddlewareExcludedRoutes(hs.GetVDBArchivedCheckMiddleware, archivalRoutes, nil))
+		internalDashboardRouter.Use(hs.VDBAuthMiddleware, hs.GetMiddlewareExcludedRoutes(hs.GetVDBArchivedCheckMiddleware, archivalRoutes, nil))
```

Comment:
Addendum to my last comment, you can just do this here:
```suggestion
		publicDashboardRouter.Use(hs.VDBAuthMiddleware, hs.ManageViaApiCheckMiddleware)
		internalDashboardRouter.Use(hs.VDBAuthMiddleware)
		// create new subrouters for archived check middleware, will be used for all endpoints added after this
		publicDashboardRouter = publicDashboardRouter.NewRoute().Subrouter()
		internalDashboardRouter = internalDashboardRouter.NewRoute().Subrouter()
		publicDashboardRouter.Use(hs.VDBArchivedCheckMiddleware)
		internalDashboardRouter.Use(hs.VDBArchivedCheckMiddleware)
```